mp/src/game/client/momentum/ui/menus/hud_menu_static_system.cpp

```diff
@@ -0,0 +1,135 @@
+#include "cbase.h"
+
+#include "hud_menu_static_system.h"
+#include "hud_menu_static.h"
+#include "filesystem.h"
+
+#include "tier0/memdbgon.h"
+
+CON_COMMAND(mom_hud_menu_reload, "Reloads the menu listing. These are located in cfg/menus/")
+{
+    g_pHudMenuStaticSystem->ReloadMenuList();
+}
+
+CON_COMMAND(mom_hud_menu_hide, "Hides the currently open hud menu if there is one.")
+{
+    g_pHudMenuStaticSystem->HideMenu();
+}
+
+static int HudMenuCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
+{
+    const auto pCmdName = "mom_hud_menu_show";
+    char *pSubstring = nullptr;
+    if (Q_strstr(pPartial, pCmdName) && strlen(pPartial) > strlen(pCmdName) + 1)
+    {
+        pSubstring = (char *)pPartial + strlen(pCmdName) + 1;
+    }
+
+    const bool bSubstringNullOrEmpty = !pSubstring || !pSubstring[0];
+    int current = 0;
+
+    for (int i = 0; i < g_pHudMenuStaticSystem->GetNumMenus(); i++)
+    {
+        const char *menuName = g_pHudMenuStaticSystem->GetMenuName(i);
+
+        if (bSubstringNullOrEmpty || Q_stristr(menuName, pSubstring))
+        {
+            char command[COMMAND_COMPLETION_ITEM_LENGTH];
+            Q_snprintf(command, COMMAND_COMPLETION_ITEM_LENGTH, "%s %s", pCmdName, menuName);
+            Q_strncpy(commands[current], command, COMMAND_COMPLETION_ITEM_LENGTH);
+            current++;
+        }
+    }
+
+    return current;
+}
+
+CON_COMMAND_F_COMPLETION(mom_hud_menu_show, "Opens a hud menu. Accepts the hud menu itself by file name. See cfg/menus/", 0, HudMenuCompletion)
+{
+    if (args.ArgC() > 1)
+    {
+        g_pHudMenuStaticSystem->ShowMenu(args[1]);
+    }
+    else
+    {
+        Warning("This command expects a parameter that is the hud menu you want to open, by file name");
```

Comment:
Missing newline

Reply:
I was missing them in a lot of places. Added them now.